app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -53,17 +53,21 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
 
     new_records = []
 
-    hashes.each do |h|
-      found = save_inventory_with_findkey(association, h.except(*remove_keys), deletes_index, new_records, record_index)
-      save_child_inventory(found, h, child_keys)
+    ActiveRecord::Base.transaction do
+      hashes.each do |h|
+        found = save_inventory_with_findkey(association, h.except(*remove_keys), deletes_index, new_records, record_index)
+        save_child_inventory(found, h, child_keys)
+      end
     end
 
     # Delete the items no longer found
     deletes = deletes_index.values
-    unless deletes.blank?
-      type = association.proxy_association.reflection.name
-      _log.info("[#{type}] Deleting #{log_format_deletes(deletes)}")
-      disconnect ? deletes.each(&:disconnect_inv) : association.delete(deletes)
+    ActiveRecord::Base.transaction do
+      unless deletes.blank?
+        type = association.proxy_association.reflection.name
+        _log.info("[#{type}] Deleting #{log_format_deletes(deletes)}")
+        disconnect ? deletes.each(&:disconnect_inv) : association.delete(deletes)
+      end
```

Comment:
If there are no deletes, doesn't this introduce an empty BEGIN/COMMIT pair because of the transaction?  Instead, is it better to do

```ruby
  unless deletes.blank?
    ActiveRecord::Base.transaction do
      type = association.proxy_association.reflection.name
      _log.info("[#{type}] Deleting #{log_format_deletes(deletes)}")
      disconnect ? deletes.each(&:disconnect_inv) : association.delete(deletes)
    end
  end